tests/models/univnet/test_modeling_univnet.py

```diff
@@ -0,0 +1,192 @@
+        inputs_dict = {"spectrogram": spectrogram, "noise_waveform": noise_waveform}
+        return config, inputs_dict
+
+
+@require_torch
+class UnivNetGanTest(ModelTesterMixin, unittest.TestCase):
+    all_model_classes = (UnivNetGan,) if is_torch_available() else ()
+    test_torchscript = False
+    test_pruning = False
+    test_resize_embeddings = False
+    test_resize_position_embeddings = False
+    test_head_masking = False
+    test_mismatched_shapes = False
+    test_missing_keys = False
+    test_model_parallel = False
+    is_encoder_decoder = False
+    has_attentions = False
+
+    input_name = "spectrogram"
+
+    def setUp(self):
+        self.model_tester = UnivNetGanTester(self)
+        self.config_tester = ConfigTester(self, config_class=UnivNetGanConfig)
+
+    def test_config(self):
+        self.config_tester.create_and_test_config_to_json_string()
+        self.config_tester.create_and_test_config_to_json_file()
+        self.config_tester.create_and_test_config_from_and_save_pretrained()
+        self.config_tester.create_and_test_config_from_and_save_pretrained_subfolder()
+        self.config_tester.create_and_test_config_with_num_labels()
+        self.config_tester.check_config_can_be_init_without_params()
+        self.config_tester.check_config_arguments_init()
+
+    def test_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model(*config_and_inputs)
+
+    def test_forward_signature(self):
+        config, _ = self.model_tester.prepare_config_and_inputs_for_common()
+
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+            signature = inspect.signature(model.forward)
+            # signature.parameters is an OrderedDict => so arg_names order is deterministic
+            arg_names = [*signature.parameters.keys()]
+
+            expected_arg_names = [
+                "spectrogram",
+            ]
+            self.assertListEqual(arg_names[: len(expected_arg_names)], expected_arg_names)
+
+    # this model does not output hidden states
+    def test_hidden_states_output(self):
+        pass
+
+    # skip
+    def test_initialization(self):
+        pass
+
+    # this model has no inputs_embeds
+    def test_inputs_embeds(self):
+        pass
+
+    # this model has no input embeddings
+    def test_model_common_attributes(self):
+        pass
+
+    # skip as this model doesn't support all arguments tested
+    def test_model_outputs_equivalence(self):
+        pass
+
+    # this model does not output hidden states
+    def test_retain_grad_hidden_states_attentions(self):
+        pass
+
+    # skip because it fails on automapping of SpeechT5HifiGanConfig
+    def test_save_load_fast_init_from_base(self):
+        pass
+
+    # skip because it fails on automapping of SpeechT5HifiGanConfig
+    def test_save_load_fast_init_to_base(self):
+        pass
```

Comment:
Yes, I copied this testing code from `SpeechT5HifiGan`. I will verify the reasons and add `@unittest.skip` decorators as appropriate.

Reply:
I have looked over the skipped tests in `UnivNetGanTest` which are inherited from `ModelTesterMixin` (as in the code snippet above), skipping the tests for which there is a reason to (also documented now) and no longer skipping tests that pass.

The one test I'm not sure about skipping is [`ModelTesterMixin.test_initialization`](https://github.com/huggingface/transformers/blob/4a27c13f1eee26393d60d381e500e1a61970e8ee/tests/test_modeling_common.py#L443); `UnivNetGan._init_weights` initializes the `weight`s via `module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)`, which is the same as [`SpeechT5HifiGan`](https://github.com/huggingface/transformers/blob/main/src/transformers/models/speecht5/modeling_speecht5.py#L3178) and (to the best of my knowledge) most transformer models (e.g. [GPT2](https://github.com/huggingface/transformers/blob/main/src/transformers/models/gpt2/modeling_gpt2.py#L456)). So it seems like this test could pass, since presumably it passes for e.g. GPT2, but when I try the test locally it fails with

```
AssertionError: -6.326800212264061e-05 not found in [0.0, 1.0] : Parameter conv_pre.weight of model <class 'transformers.models.univnet.modeling_univnet.UnivNetGan'> seems not properly initialized
```

`UnivNetGan.conv_pre` is a `torch.nn.Conv1d`, so it should be initialized via `module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)`. [The [reference implementation](https://github.com/dg845/univnet/blob/master/model/generator.py#L9) uses PyTorch's default initialization, which I believe is He initialization for both [`torch.nn.Linear`](https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/linear.py#L103) and [`torch.nn.Conv1d`](https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/conv.py#L143), but I assume `module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)` is preferred for `transformers`.]

Reply:
You should probably do a proper call to `init_weights` [here](https://github.com/huggingface/transformers/blob/06bc28032a09cd36222682c9385033b33920ac65/src/transformers/models/univnet/modeling_univnet.py#L587).

Transformers models do it by adding a `self.post_init()` at the end of the file.
See [speecht5](https://github.com/huggingface/transformers/blob/06bc28032a09cd36222682c9385033b33920ac65/src/transformers/models/speecht5/modeling_speecht5.py#L1308-L1309) for an example but you can find other examples by skimming through examples!